src/EnergyPlus/PlantLoopEquip.cc

```diff
@@ -308,19 +307,19 @@ namespace PlantLoopEquip {
 		//Pipe has no special types at the moment, so find it this way
 		if ( GeneralEquipType == GenEquipTypes_Pipe ) {
 			if ( EquipTypeNum == TypeOf_Pipe ) {
```

Comment:
Isn't have of the point of doing this base class thing to introduce polymorphism so that you don't have these conditional switches? Can you not reduce this to one call to the base simulate?